Bugfix orders and accounts

kb0rg opened the pull request:
Addresses a few issues
 - removes cached pyc files
 - fixes [this bug](https://github.com/tangocard/raas-v2-sdk-python/issues/5), which threw an exception when trying to get all customers from the customers controller (if any accounts exist)
 - makes datetime args optional (though I'm not sure if doing this globally is a good idea or if it could break anything else -- I'll add more detailed info on that line)

kb0rg committed:
Remove cached pyc files

kb0rg committed:
Fix bug calling account_summary_model

kb0rg committed:
Make datetime optional

kb0rg commented on raas_v2/api_helper.py:
```diff
@@ -310,7 +310,8 @@ class RFC3339DateTime(CustomDate):
 
         @classmethod
         def from_datetime(cls, date_time):
```
While using [a method documented as having _optional_ inputs](https://github.com/tangocard/raas-v2-sdk-python/blob/91cd237b7921756eb6178d0b7e3703f4b943c7d7/raas_v2/controllers/orders_controller.py#L197) I hit errors if I did not provided a datetime for the "optional" startDate (see traceback below).

Am wondering though if it's safe to make this a global change, or if I should subclass the RFC3339DateTime object to make an OptionalRFC3339DateTime with this conditional?

```
INFO:raas_v2.controllers.orders_controller:get_orders called.
INFO:raas_v2.controllers.orders_controller:Preparing query URL for get_orders.
ERROR:raas_v2.controllers.orders_controller:'NoneType' object has no attribute 'isoformat'
Traceback (most recent call last):
  File "/Users/kborg/dev/__misc/temp_tango/env/lib/python2.7/site-packages/raas_v2/controllers/orders_controller.py", line 239, in get_orders
    'startDate': APIHelper.RFC3339DateTime(options.get('start_date', None)),
  File "/Users/kborg/dev/__misc/temp_tango/env/lib/python2.7/site-packages/raas_v2/api_helper.py", line 264, in __init__
    self.value = self.from_datetime(dtime)
  File "/Users/kborg/dev/__misc/temp_tango/env/lib/python2.7/site-packages/raas_v2/api_helper.py", line 313, in from_datetime
    return date_time.isoformat()
AttributeError: 'NoneType' object has no attribute 'isoformat'
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-31-fe516414b137> in <module>()
----> 1 orders_client.get_orders({'account_identifier': 'fold-dev-inventory'})

/Users/kborg/dev/__misc/temp_tango/env/lib/python2.7/site-packages/raas_v2/controllers/orders_controller.pyc in get_orders(self, options)
    237                 'customerIdentifier': options.get('customer_identifier', None),
    238                 'externalRefID': options.get('external_ref_id', None),
--> 239                 'startDate': APIHelper.RFC3339DateTime(options.get('start_date', None)),
    240                 'endDate': APIHelper.RFC3339DateTime(options.get('end_date', None)),
    241                 'elementsPerBlock': options.get('elements_per_block', None),

/Users/kborg/dev/__misc/temp_tango/env/lib/python2.7/site-packages/raas_v2/api_helper.pyc in __init__(self, dtime, value)
    262             self.datetime = dtime
    263             if not value:
--> 264                 self.value = self.from_datetime(dtime)
    265             else:
    266                 self.value = value

/Users/kborg/dev/__misc/temp_tango/env/lib/python2.7/site-packages/raas_v2/api_helper.pyc in from_datetime(cls, date_time)
    311         @classmethod
    312         def from_datetime(cls, date_time):
--> 313             return date_time.isoformat()
    314 
    315         @classmethod

AttributeError: 'NoneType' object has no attribute 'isoformat'
```

Shadowfiend replied:
Usually I'd look at the generator itself to see if they've resolved the bug on the generation side… But the generator seems like it's probably closed source, so it's impossible to know for sure. Given the Tango folks haven't touched this repo for almost a year, I would go ahead and tweak the code directly. If they do make updates, when we pull them in we can figure out how we want to handle changes.

Tbh I would make the change in the invocation to `RFC3339DateTime` though. It seems like `APIHelper.RFC3339DateTime(options.get('start_date', None))` should be something like `options.get('start_date', None) and APIHelper.RFC3339DateTime(options.get('start_date', None))`, so that the `None` short-circuits and if we have a value we parse properly.

kb0rg replied:
> I would make the change in the invocation

yes! that makes sense, will do that. thanks.

kb0rg committed:
Make datetime optional in get_orders

Revert change in api helper

kb0rg commented:
@Shadowfiend I updated with your suggested fix (and reverted the change I'd made in the api helper).

Meanwhile, I noticed that tango's generated code branch got updated yesterday. Looks like it fixes the first issue (broken import).

They haven't pulled the update into master yet, so I opened a new issue to address the datetime problem:
https://github.com/tangocard/raas-v2-sdk-python/issues/6

What's your opinion re merging vs abandoning this PR?

(Personally, my inclination is to keep this fork alive but open to discussion)
cc @mhluongo

Shadowfiend commented:
100% keep the fork. It doesn't really cost us much, we can always point our package manager back at upstream when they catch up.

Also… They're doing this on a branch other than `master` which has commits from last June that haven't been merged, as well… So I'm not sure when they're planning on merging their changes back in? Hard to know for sure. Or maybe they're publishing their package from the branch instead of `master`? <_<

arcdigital commented:
Hey, sorry for the confusion here - The Python SDK hasn't gotten an update in a while and definitely needs an update. We plan to update it (and all of our other ones) tomorrow / push the new version to PyPI. Going forward if you have any other issues, feel free open an issue on GitHub and we'll respond :)

Shadowfiend commented:
:tada: It's not so hard to know for sure when you're kind enough to share with us :) Thanks for the response, @arcdigital !

kb0rg commented:
Issues fixed in [the latest tangocard sdk release](https://github.com/tangocard/raas-v2-sdk-python/pull/7)
